copt/frank_wolfe.py

```diff
@@ -205,11 +211,11 @@ def minimize_frank_wolfe(
 
         if certificate <= tol:
             break
-        if hasattr(step_size, "__call__"):
-            step_size_t = step_size(locals())
-            f_next, grad_next = f_grad(x + step_size_t * update_direction)
-        elif step_size == "adaptive":
-            step_size_t, lipschitz_t, f_next, grad_next = adaptive_step_size(
+        if hasattr(step, "__call__"):
+            step_size = step(locals())
```

Comment:
I'm new to using `locals()` in this way. But it's an interesting way of allowing functions with arbitrary function signatures. Could you also use `**kwargs`?

Reply:
We could pass it as **locals() and then the trace would have keyword arguments instead of a single dictionary as input. I don't have a strong opinion either way

Reply:
Me neither. I think it's fine to leave it as is.